specification/synapse/data-plane/Microsoft.Synapse/preview/2019-06-01-preview/entityTypes/Dataset.json

```diff
@@ -612,17 +753,19 @@
           "description": "The code page name of the preferred encoding. If miss, the default value is UTF-8, unless BOM denotes another Unicode encoding. Refer to the name column of the table in the following link to set supported values: https://msdn.microsoft.com/library/system.text.encoding.aspx. Type: string (or Expression with resultType string)."
         },
         "compressionCodec": {
-          "type": "string",
+          "type": "object",
```

Comment:
What is the reason for this string changing to an object?

Reply:
Hi, we are syncing with the Azure Data Factory(ADF) swagger. The Synapse-side swagger is out of date with Azure Data Factory counterpart. This has led to a lot feature gaps Synapse customers.

Reply:
Here is the link to the ADF public swagger 
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json

Reply:
This was an incorrect change in ADF in the first place, this is not an object. Sorry, but we should not just accept it here because it's copy/pasting :(. We shoudl fix ADF Swagger instead.

Reply:
Agree. For enumeration, the type should be string. During code generation, I can see some warnings if type if object. @drosales57 , could you please at least fix this on Synapse side?